server/src/main/java/org/opensearch/cluster/routing/RecoverySource.java

```diff
@@ -415,10 +415,10 @@ public void addAdditionalFields(XContentBuilder builder, ToXContent.Params param
                 .field("version", version.toString())
                 .field("index", index.getName())
                 .field("restoreUUID", restoreUUID)
-                .field("isSearchableSnapshot", isSearchableSnapshot)
-                .field("remoteStoreIndexShallowCopy", remoteStoreIndexShallowCopy)
-                .field("sourceRemoteStoreRepository", sourceRemoteStoreRepository)
-                .field("sourceRemoteTranslogRepository", sourceRemoteTranslogRepository);
+                .field("is_searchable_snapshot", is_searchable_snapshot)
```

Comment:
Java variables should follow the lowerCamelCase convention, but when serialized in `toXContent` (as done on the left side of this line), these can be converted to lower_snake_case.

As @gbbafna pointed out [here](https://github.com/opensearch-project/OpenSearch/issues/16334#issuecomment-2419851207), since this is a change in API its considered a breaking change and would have to be targeted for OpenSearch 3.0. At the moment, OpenSearch 3.0 is the next planned release after OpenSearch 2.19.